### PR TITLE
DM-483 permissions update to SWU opps

### PR DESCRIPTION
**Pull request by sutherlanda**

### CODE REVIEW

File: src/back-end/lib/permissions.ts [hunk 1]
```diff
@@ -227,8 +227,21 @@ export function publishSWUOpportunity(session: Session): boolean {
   return isAdmin(session);
 }
 
-export async function deleteSWUOpportunity(connection: Connection, session: Session, opportunityId: string): Promise<boolean> {
-  return isAdmin(session) || (session && isGovernment(session) && await isSWUOpportunityAuthor(connection, session.user, opportunityId)) || false;
+export async function deleteSWUOpportunity(connection: Connection, session: Session, opportunityId: string, status: SWUOpportunityStatus): Promise<boolean> {
+  return isAdmin(session) && [SWUOpportunityStatus.Draft, SWUOpportunityStatus.UnderReview].includes(status)
```
Comment (dhruvio) [inline src/back-end/lib/permissions.ts:231]: Can you please add parentheses around the admin check before the `||` operator for legibility?

File: src/back-end/lib/permissions.ts [hunk 2]
```diff
@@ -227,8 +227,21 @@ export function publishSWUOpportunity(session: Session): boolean {
...
+    || (session && isGovernment(session) && await isSWUOpportunityAuthor(connection, session.user, opportunityId) && status === SWUOpportunityStatus.Draft) || false;
```
Comment (dhruvio) [inline src/back-end/lib/permissions.ts:232]: You don't need the `session &&` bit

File: src/back-end/lib/permissions.ts (hunk 1 above)
Comment (sutherlanda) [reply src/back-end/lib/permissions.ts]: Done

File: src/back-end/lib/permissions.ts (hunk 2 above)
Comment (sutherlanda) [reply src/back-end/lib/permissions.ts]: It throws a type error on the following call if I don't due to TypeScript weirdness.  I've changed it to use isSignedIn() for consistency.

**Pull request merged**